### DISCUSSION TITLE
Version 2.1.0

### DISCUSSION THREAD

**Discussion by samerton** [NamelessMC/Nameless#3347] [Releases]
### Added
- Add dark mode toggle switch [#2877](https://github.com/NamelessMC/Nameless/pull/2877)
- Add option to use OAuth linking method for Discord Integration [#3051](https://github.com/NamelessMC/Nameless/pull/3051)
- Add limit setting to latest posts widget [#2862](https://github.com/NamelessMC/Nameless/issues/2862), [#3107](https://github.com/NamelessMC/Nameless/pull/3107)
- Add option for user to reset their avatar [#3042](https://github.com/NamelessMC/Nameless/issues/3042), [#3100](https://github.com/NamelessMC/Nameless/pull/3100)
- Add integrations settings system & Setting to turn off mc username requirement [#3043](https://github.com/NamelessMC/Nameless/issues/3043), [#3109](https://github.com/NamelessMC/Nameless/pull/3109)
- Add semantic config and change node_modules copy method [#3111](https://github.com/NamelessMC/Nameless/pull/3111)
- Add createWebhookEndpoint [#3096](https://github.com/NamelessMC/Nameless/pull/3096)
- Add new webhook events for profile posts and profile post replies [#3073](https://github.com/NamelessMC/Nameless/pull/3073)
- Add Finnish language [#3117](https://github.com/NamelessMC/Nameless/pull/3117)
- Add groups parameter to list users endpoint [#3052](https://github.com/NamelessMC/Nameless/issues/3052), [#3193](https://github.com/NamelessMC/Nameless/pull/3193)
- Add support for automatically verifying emails from OAuth register [#3203](https://github.com/NamelessMC/Nameless/pull/3203)
- Add ability to query servers via plugin [#3244](https://github.com/NamelessMC/Nameless/pull/3244)
- Allow selecting between native/twemoji/emojione emojis [#3269](https://github.com/NamelessMC/Nameless/pull/3269)
- Add Latvian language [#3277](https://github.com/NamelessMC/Nameless/pull/3277)
- Create Members module with member list page [#3106](https://github.com/NamelessMC/Nameless/pull/3106)
- Allow enabling/disabling modules during installer [#3273](https://github.com/NamelessMC/Nameless/pull/3273)
- Add delete button for placeholders [#3069](https://github.com/NamelessMC/Nameless/issues/3069), [#3283](https://github.com/NamelessMC/Nameless/pull/3283)
- Queue system [#3274](https://github.com/NamelessMC/Nameless/pull/3274)
- Add support to customise OAuth login buttons [#3285](https://github.com/NamelessMC/Nameless/pull/3285)
- Add ability to customise number of news items on front page [#3303](https://github.com/NamelessMC/Nameless/pull/3303)
- Create `HasWebhookParams` interface [#3275](https://github.com/NamelessMC/Nameless/issues/3275), [#3281](https://github.com/NamelessMC/Nameless/pull/3281)
- Integrity checking system [#3159](https://github.com/NamelessMC/Nameless/pull/3159)
- Cancellable events [#3329](https://github.com/NamelessMC/Nameless/pull/3329)
- Add dependency injection library + task to generate sitemap [#3332](https://github.com/NamelessMC/Nameless/pull/3332)

### Changed
- Handle OAuth errors [#3030](https://github.com/NamelessMC/Nameless/pull/3030)
- Allow OAuth providers to pass additional options to provider constructor [#3062](https://github.com/NamelessMC/Nameless/pull/3062)
- Place dark mode toggle inline [#3054](https://github.com/NamelessMC/Nameless/pull/3054)
- Use composer platform php version [#3063](https://github.com/NamelessMC/Nameless/pull/3063)
- Convert error page to Fomantic UI [#3029](https://github.com/NamelessMC/Nameless/pull/3029)
- Update TFA page to use a post request to disable TFA [#3080](https://github.com/NamelessMC/Nameless/pull/3080)
- Show message if there was an error loading a widget [#3138](https://github.com/NamelessMC/Nameless/pull/3138)
- Clarify meaning of "query" [#3142](https://github.com/NamelessMC/Nameless/pull/3142)
- Change exception page Prism theme to One Light [#3163](https://github.com/NamelessMC/Nameless/pull/3163)
- Update Prism token colours [#3169](https://github.com/NamelessMC/Nameless/pull/3169)
- Catch exception when installing modules [#2875](https://github.com/NamelessMC/Nameless/issues/2875), [#3143](https://github.com/NamelessMC/Nameless/pull/3143)
- Remove group sync info from user info endpoint [#3194](https://github.com/NamelessMC/Nameless/pull/3194)
- PHPDoc: Use debian image and speed up build [#3201](https://github.com/NamelessMC/Nameless/pull/3201)
- Check for pdo_mysql extension, mysqli is no longer used [#3202](https://github.com/NamelessMC/Nameless/pull/3202)
- Remove panel template "Find Templates" section [#3196](https://github.com/NamelessMC/Nameless/pull/3196)
- Small email tidy up [#3215](https://github.com/NamelessMC/Nameless/pull/3215)
- Swap API URL and API key fields [#3218](https://github.com/NamelessMC/Nameless/pull/3218)
- AuthMe improvements and fixes [#3198](https://github.com/NamelessMC/Nameless/pull/3198)
- Update server status page to refresh every 5 seconds [#3223](https://github.com/NamelessMC/Nameless/pull/3223)
- Remove status page cache [#3228](https://github.com/NamelessMC/Nameless/pull/3228)
- Remove MCAssoc [#3220](https://github.com/NamelessMC/Nameless/pull/3220)
- Don't validate unused parameters [#3235](https://github.com/NamelessMC/Nameless/pull/3235)
- Rewrite Discord group sync to use v5 Nameless-Link API [#2945](https://github.com/NamelessMC/Nameless/issues/2945), [#3222](https://github.com/NamelessMC/Nameless/pull/3222)
- Update contributing documentation to mention npm [#3242](https://github.com/NamelessMC/Nameless/pull/3242)
- Remove query interval cache [#3247](https://github.com/NamelessMC/Nameless/pull/3247)
- Attempt to remove roles in Discord when unlinking Discord integration [#3236](https://github.com/NamelessMC/Nameless/pull/3236)
- Remove MINECRAFT constant [#3225](https://github.com/NamelessMC/Nameless/pull/3225)
- Remove code scheduled for removal in 2.1.0 [#3165](https://github.com/NamelessMC/Nameless/pull/3165)
- Misc fixes and improvements [#3265](https://github.com/NamelessMC/Nameless/issues/3265), [#3267](https://github.com/NamelessMC/Nameless/issues/3267), [#2861](https://github.com/NamelessMC/Nameless/issues/2861), [#3270](https://github.com/NamelessMC/Nameless/pull/3270), [#3284](https://github.com/NamelessMC/Nameless/pull/3284)
- Make sure the dark mode switcher has a pointer cursor [#3289](https://github.com/NamelessMC/Nameless/pull/3289)
- Make sure that installation has finished [#3297](https://github.com/NamelessMC/Nameless/pull/3297)
- Use labels instead of code within installer [#3296](https://github.com/NamelessMC/Nameless/pull/3296)
- Make sure the Like button is at the bottom [#2892](https://github.com/NamelessMC/Nameless/issues/2892), [#3305](https://github.com/NamelessMC/Nameless/pull/3305)
- Use regex to determine which files to treat as migration files [#3287](https://github.com/NamelessMC/Nameless/issues/3287), [#3308](https://github.com/NamelessMC/Nameless/pull/3308)
- Better handling of closures in exception frames [#3309](https://github.com/NamelessMC/Nameless/pull/3309)
- Remove the v1 converter during installation [#3293](https://github.com/NamelessMC/Nameless/pull/3293)
- Deprecate codeTransform and decode methods within ContentHook [#3250](https://github.com/NamelessMC/Nameless/pull/3250)
- Don't decode before encoding for new installations [#3171](https://github.com/NamelessMC/Nameless/pull/3171)
- Do not encode welcome post [#3316](https://github.com/NamelessMC/Nameless/pull/3316)
- Use relative paths so phpdoc treats it as a single docset [#3318](https://github.com/NamelessMC/Nameless/issues/3318), [#3324](https://github.com/NamelessMC/Nameless/pull/3324)
- Use utf8mb4_unicode_ci for DB class to match Phinx [#3179](https://github.com/NamelessMC/Nameless/issues/3179), [#3317](https://github.com/NamelessMC/Nameless/pull/3317)
- Setting constants [#3232](https://github.com/NamelessMC/Nameless/pull/3232)
- Replace manual queries with util setting functions [#3224](https://github.com/NamelessMC/Nameless/pull/3224)
- Improve release script [#3325](https://github.com/NamelessMC/Nameless/pull/3325)
- Allow correct webhook action types in endpoint [#3326](https://github.com/NamelessMC/Nameless/pull/3326)
- Use CSS instead of the script to put the footer at the bottom [#3301](https://github.com/NamelessMC/Nameless/pull/3301)
- Make sure module is enabled to process task [#3331](https://github.com/NamelessMC/Nameless/pull/3331)
- Initialise tasks in installer [#3342](https://github.com/NamelessMC/Nameless/pull/3342)
- Dependency updates
- Translation updates [#3252](https://github.com/NamelessMC/Nameless/pull/3252)

### Fixed
- Fix private profiles [#3020](https://github.com/NamelessMC/Nameless/pull/3020)
- Fix for some pages which don't show the updated values/data [#3105](https://github.com/NamelessMC/Nameless/pull/3105)
- Fix Discord verify command translation [#3087](https://github.com/NamelessMC/Nameless/pull/3087)
- Fix hardcoded language terms [#3036](https://github.com/NamelessMC/Nameless/issues/3036), [#3113](https://github.com/NamelessMC/Nameless/pull/3113)
- Fix ROOT_PATH in cli installer [#3123](https://github.com/NamelessMC/Nameless/pull/3123)
- Use X-Real-IP header last [#3230](https://github.com/NamelessMC/Nameless/pull/3230)
- Fix `register_method` column not being filled when accounts are created [#3231](https://github.com/NamelessMC/Nameless/pull/3231)
- Add smarty.js to postinstall script [#3245](https://github.com/NamelessMC/Nameless/pull/3245)
- Fix member module issues [#3282](https://github.com/NamelessMC/Nameless/pull/3282)
- Fix permissions not being respected for news posts [#3290](https://github.com/NamelessMC/Nameless/pull/3290)
- Fix errors during admin creation [#3295](https://github.com/NamelessMC/Nameless/pull/3295)
- Fix nofollow noopener on external links [#3306](https://github.com/NamelessMC/Nameless/pull/3306)
- Make like button text not hardcoded [#3304](https://github.com/NamelessMC/Nameless/pull/3304)
- Recaptcha v3 validation not working [#3276](https://github.com/NamelessMC/Nameless/issues/3276), [#3307](https://github.com/NamelessMC/Nameless/pull/3307)
- Fix missing event in regular webhook [#3327](https://github.com/NamelessMC/Nameless/pull/3327)
- Fix default user template setting [#3330](https://github.com/NamelessMC/Nameless/pull/3330)
- Add missing avatar upload directory [#3333](https://github.com/NamelessMC/Nameless/pull/3333)
- Fix cookie text on StaffCP permissions page [#3336](https://github.com/NamelessMC/Nameless/pull/3336)
- Fix DataCollector not found error [#3341](https://github.com/NamelessMC/Nameless/pull/3341)

## New Contributors
* @RageOfFire made their first contribution in https://github.com/NamelessMC/Nameless/pull/3026
* @Browsit made their first contribution in https://github.com/NamelessMC/Nameless/pull/3068
* @Co0kei made their first contribution in https://github.com/NamelessMC/Nameless/pull/3073

**Full Changelog**: https://github.com/NamelessMC/Nameless/compare/v2.0.3...v2.1.0

---

### Which zip should I download?
- If you are updating from 2.0.3, please follow the instructions in your StaffCP -> Update tab.
- If you are installing a new website, download `nameless-deps-dist.zip` or `nameless-deps-dist.tar.xz` which will include all the files you need for a fresh start.

<hr /><em>This discussion was created from the release <a href='https://github.com/NamelessMC/Nameless/releases/tag/v2.1.0'>Version 2.1.0</a>.</em>

**Comment by SupaYoshi** [comment]
If, after the upgrade from 2.0.3 to 2.1.0, the site redirects to install.php, you should modify the /core/config.php file. Add the line 'installed' => true, under the core section as follows:

php

'core' => [
    'path' => '',
    'hostname' => 'yourwebsite.com',
    'friendly' => true,
    'force_https' => true,
    'force_www' => false,
    'installed' => true,
],

**Comment by SupaYoshi** [comment]
**### Ensure you have a backup before proceeding with these steps. This will help you restore your system in case something goes wrong during the upgrade process.**

Upgrade Guide: NamelessMC from 2.0.3 to 2.1.0

If you're encountering issues while upgrading from NamelessMC version 2.0.3 to 2.1.0, follow these steps:

    Modify core/init.php:
        Open the file core/init.php.
        Go to line 110 and add PhinxAdapter::migrate(); above this line.

    Update modules/Core/pages/panel/upgrade.php:
        Open the file modules/Core/pages/panel/upgrade.php.
        Comment out lines 13 to 17 as the bot suggests.

    Check the Site and Error Log:
        After making the changes, check if the site is loading.
        If the site is not loading, open the error log using the command tail -f cache/logs/other-log.log.
        If you see errors like "Base table or view already exists: 1050 Table 'nl2_queue' already exists" or the "nl2_member_list error" in the run log:
            Go to phpMyAdmin.
            Drop the tables nl2_queue and nl2_member_list (these tables usually don't contain much data).

    Revert Changes in core/init.php:
        Change back the file core/init.php by removing PhinxAdapter::migrate();.

    Optional: Revert Changes in modules/Core/pages/panel/upgrade.php:
        You may also remove the changes made in Step 2.

Congratulations! Your upgrade from NamelessMC version 2.0.3 to 2.1.0 should now be complete.

Ensure you have a backup before proceeding with these steps. This will help you restore your system in case something goes wrong during the upgrade process.
